### PR TITLE
Split GetCredential into authentication and role switching

### PR DESCRIPTION
**Pull request by leuyentran**
## Change Overview

Split AWS `GetCredentials` into smaller functions:

1. `authenticateAWSCredentials` : authenticate AWS IAM identity either by credentials or via injected token
2. `awsRoleSwitching`:  after authentication, assume another role if available and allowed 

The refactor provides two benefits:

1. Readability 
2. Match with K10 installation doc
3. Extendibility: more authentication methods can be added in the future if desired 

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Test Plan

For good measure, tested validation of an AWS infra profile which calls `GetCredentials` 

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/aws/aws.go [hunk 1]
```diff
@@ -68,43 +68,62 @@ func durationFromString(config map[string]string) (time.Duration, error) {
...
 	}
-	// If the caller didn't want to assume a different role, we're done
-	if config[ConfigRole] == "" || config[ConfigRole] == assumedRole {
+	return creds, assumedRole, nil
+}
+
+// check of caller want to assume a different role
+// return as is if ConfigRole is empty, or already same as assumedRole
+// otherwise proceed to switch role
+func awsRoleSwitching(ctx context.Context, creds *credentials.Credentials, targetRole string, currentRole string, assumeRoleDuration time.Duration) (*credentials.Credentials, error) {
```
Comment (DeepikaDixit) [inline pkg/aws/aws.go:95]: Rename: `switchAwsRole`

**Review by DeepikaDixit — COMMENTED** [review]
Will let @pavannd1 have a look

File: pkg/aws/aws.go (hunk 1 above)
Comment (leuyentran) [reply pkg/aws/aws.go]: done!

File: pkg/aws/aws.go [hunk 2]
```diff
@@ -68,43 +68,62 @@ func durationFromString(config map[string]string) (time.Duration, error) {
...
 		sess, err := session.NewSessionWithOptions(session.Options{AssumeRoleDuration: assumeRoleDuration})
 		if err != nil {
-			return nil, errors.Wrap(err, "Failed to create session to initialize Web Identify credentials")
+			return nil, "", errors.Wrap(err, "Failed to create session to initialize Web Identify credentials")
 		}
 		creds = getCredentialsWithDuration(sess, assumeRoleDuration)
 		assumedRole = os.Getenv(roleARNEnvKey)
 	default:
-		return nil, errors.New("AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY required to initialize AWS credentials")
+		return nil, "", errors.New("AWS_ACCESS_KEY_ID and AWS_3SECRET_ACCESS_KEY required to initialize AWS credentials")
```
Comment (miquella) [inline pkg/aws/aws.go:87]:
```suggestion
		return nil, "", errors.New("AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY required to initialize AWS credentials")
```

Comment (pavannd1) [inline pkg/aws/aws.go]:
```suggestion
		return nil, "", errors.New("AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY required to initialize AWS credentials")
```

File: pkg/aws/aws.go [hunk 3]
```diff
@@ -68,43 +68,62 @@ func durationFromString(config map[string]string) (time.Duration, error) {
...
+		// When you use role chaining, your new credentials are limited to a maximum duration of one hour
+		// https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use.html
+		if currentRole != "" {
+			assumeRoleDuration = 60 * time.Minute
+		}
+		// If the caller wants to use a specific role, use the credentials initialized above to assume that
+		// role and return those credentials instead
+		creds, err := awsrole.Switch(ctx, creds, targetRole, assumeRoleDuration)
+		return creds, errors.Wrap(err, "Failed to switch roles")
 	}
```
Comment (pavannd1) [inline pkg/aws/aws.go:108]:
```suggestion

	// When you use role chaining, your new credentials are limited to a maximum duration of one hour
	// https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use.html
	if currentRole != "" {
		assumeRoleDuration = 60 * time.Minute
	}
	// If the caller wants to use a specific role, use the credentials initialized above to assume that
	// role and return those credentials instead
	creds, err := awsrole.Switch(ctx, creds, targetRole, assumeRoleDuration)
	return creds, errors.Wrap(err, "Failed to switch roles")
```
for better readability

Comment (pavannd1) [reply pkg/aws/aws.go]: @leuyentran Oops sorry about the bracket 🙏🏼

**Review by pavannd1 — APPROVED** [review]
LGTM

**Pull request merged**